firmware/controllers/core/state_sequence.cpp

```diff
@@ -9,42 +9,11 @@
 #include "state_sequence.h"
 #include "trigger_structure.h"
 
-SingleChannelStateSequence::SingleChannelStateSequence() {
-	init(NULL);
-}
-
-SingleChannelStateSequence::SingleChannelStateSequence(pin_state_t *ps) {
-	init(ps);
-}
-
-void SingleChannelStateSequence::init(pin_state_t *pinStates) {
-	this->pinStates = pinStates;
-}
-
-pin_state_t SingleChannelStateSequence::getState(int switchIndex) const {
-	pin_state_t state = pinStates[switchIndex];
-	efiAssert(OBD_PCM_Processor_Fault, state == 0 || state == 1, "wave state get", TV_FALL);
-	return state;
-}
-
-void SingleChannelStateSequence::setState(int switchIndex, pin_state_t state) {
-	efiAssertVoid(OBD_PCM_Processor_Fault, state == 0 || state == 1, "wave state set");
-	pinStates[switchIndex] = state;
-}
-
-MultiChannelStateSequence::MultiChannelStateSequence() {
+MultiChannelStateSequence::MultiChannelStateSequence(unsigned maxWaveCount)
+	: wavePtr(reinterpret_cast<uint8_t *>(&switchTimes[maxWaveCount])) {
```

Comment:
possible to avoid `reinterpret_cast`?

Reply:
I could make a union between switchTimes and waves, but in the end you need to know to skip maxWaveCount floats, not bytes, so you're going to have some dodgy pointer math somewhere.  I thought reinterpret_cast would be the lesser of the two evils.